### PR TITLE
Feature rules delete

### PR DESCRIPTION
**Pull request by maxkimambo**
+ Added ability to associate rules to a job 
+ Extended repository to search contents of array
+ When no rules are associated with the job, the job is deleted
+ Added endpoint to delete rules

### CODE REVIEW

File: server/docs/rule-payload.json [hunk 1]
```diff
@@ -16,18 +16,27 @@
...
         "comparator": "gt",
-        "value": 25
+        "value": 20
     },
-    "executionInterval": 30,
+    "executionInterval": 60,
     "targets": [
         {
-            "agentId": "google-ads",
+            "agentId": "dv360-agent",
```
Comment (paranerd) [inline server/docs/rule-payload.json:24]: Why `"dv360-agent"` instead of just `"dv360"`?

**Review by paranerd — CHANGES_REQUESTED** [review]

File: server/agents/source-agents/open-weather-map/interfaces.ts [hunk 2]
```diff
@@ -17,20 +18,22 @@ export interface Configuration {
     name: string;
     targetLocation?: string;
     jobId?: string;
+    jobOwner: string;
 }
 
 export interface AgentResult {
     agentId: string;
     jobId: string;
+    jobOwner: string;
```
Comment (akushnarov) [inline server/agents/source-agents/open-weather-map/interfaces.ts:27]: Is the "jobId" already a pointer to the owner and we can get rid of the owner in the agents?

File: server/controllers/jobs-controller.ts [hunk 3]
```diff
@@ -71,7 +73,55 @@ export const addJob = async (rule: Rule): Promise<string> => {
...
+};
+
+/**
+ * Disassociate the rule from a job.
+ * Job is deleted if its no longer associated to any of the rules.
+ * @param {string} ruleId
+ */
+export const removeRuleFromJob = async (ruleId: string) => {
+    try {
+        const jobs = await repo.arrayContains('rules', ruleId);
```
Comment (akushnarov) [inline server/controllers/jobs-controller.ts:106]: A bit confusing: you select `rules` but the variable is `jobs`.

**Review by akushnarov — APPROVED** [review]

File: server/controllers/jobs-controller.ts (hunk 3 above)
Comment (maxkimambo) [reply server/controllers/jobs-controller.ts:106]: I dont select rules, I select jobs which have rules [] and look for ruleId in that array

**Pull request merged**